geotrek/maintenance/filters.py

```diff
@@ -157,6 +157,11 @@ class Meta(StructureRelatedFilterSet.Meta):
             'status', 'type', 'stake', 'subcontracting', 'project', 'contractors', 'on',
         ]
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Sorted 'ON_CHOICES' here to sort with user language.
+        self.form.fields["on"].choices = sorted(self.ON_CHOICES, key=lambda modelname: modelname[1])
```

Comment:
Have you tested it? (I am curious does `sorted` trigger the lazy eval of translated `gettext`?)

Reply:
yes, tested and fixed with me. sort elements in init method make sort works according current translation activated